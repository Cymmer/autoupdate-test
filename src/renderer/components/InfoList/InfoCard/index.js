import React from 'react';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

import { Card, Text, Icon, Code } from '../../elements';
import {
  textTypes,
  getSyntaxHighlighterLanguageFromExtension,
} from '../../elements/constants';

const InfoCard = ({
  title,
  info,
  syntax,
  code,
  output,
  languageExtension,
  additionalInfo,
}) => (
  <>
    <Card className={styles.InfoCard}>
      <div className={styles.InfoCard_accent} />
      <div className={styles.InfoCard_content}>
        <Text
          className={styles.InfoCard_subject}
          colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['700']}
          type={textTypes.HEADING.XS}
        >
          {title}
        </Text>
        <Text
          className={styles.InfoCard_description}
          colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['600']}
          type={textTypes.BODY.MD}
        >
          <span dangerouslySetInnerHTML={{ __html: info }} />
        </Text>
        {syntax && (
          <div className={styles.InfoCard_syntax}>
            <Text
              colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['500']}
              type={textTypes.BODY.MD}
            >
              Syntax:
            </Text>
            {syntax.map((text, index) => (
              <Code
                key={index}
                className={styles.InfoCard_syntax_code}
                language={getSyntaxHighlighterLanguageFromExtension(
                  languageExtension
                )}
              >
                {text}
              </Code>
            ))}
          </div>
        )}

        {code && (
          <div className={styles.InfoCard_code}>
            <Code
              language={getSyntaxHighlighterLanguageFromExtension(
                languageExtension
              )}
            >
              {code}
            </Code>
            {output && (
              <>
                <Text type={textTypes.HEADING.XXXS}>Output:</Text>
                <Code>{output}</Code>
              </>
            )}
          </div>
        )}
      </div>
    </Card>

    {additionalInfo && (
      <Card className={styles.InfoCard_additional}>
        <Icon className={styles.InfoCard_additional_icon} icon="info" />
        <Text className={styles.InfoCard_additional_text}>
          {additionalInfo}
        </Text>
      </Card>
    )}
  </>
);

InfoCard.defaultProps = {
  syntax: null,
  output: null,
  additionalInfo: null,
  code: null,
  languageExtension: null,
};

InfoCard.propTypes = {
  title: PropTypes.string.isRequired,
  info: PropTypes.string.isRequired,

  // an array of strings that contains all the possible
  // syntaxes for this current info
  syntax: PropTypes.arrayOf(PropTypes.string),

  // an actual sample code for this current info
  code: PropTypes.string,

  // the output of the actual sample code. This only
  // contains some value if there is a sample code.
  output: PropTypes.string,

  // the programming language set for this editor
  languageExtension: PropTypes.oneOf(GLOBALS.LANGUAGE_EXTENSIONS_PROP_TYPES),

  // any additional information
  additionalInfo: PropTypes.string,
};

export default InfoCard;
