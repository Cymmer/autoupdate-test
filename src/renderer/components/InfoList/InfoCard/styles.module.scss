@import '../../../scss/variables';
@import '../../../scss/mixins';

.InfoCard {
  display: flex;
  overflow: hidden;

  &_accent {
    width: 16px;
    background-color: $blue-100;
  }

  &_content {
    padding: 24px 32px;
    width: calc(100% - 16px);

    @include breakpoint(sm) {
      padding: 24px;
    }
  }

  &_subject {
    margin: 0 0 12px 0;
  }

  &_additional {
    padding: 24px;
    display: flex;
    align-items: center;
    background-color: lighten($yellow-100, 5%) !important;

    &_icon {
      font-size: 2rem;
      color: $yellow-300;
      margin: 0 32px 0 0;
    }

    &_text {
      color: $neutral-700 !important;
    }
  }

  &_syntax {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    margin: 12px 0 0 0;

    &_code {
      margin: 4px;
      width: max-content;
      overflow: auto;
    }
  }

  &_code {
    margin: 16px 0 0 0;
    display: grid;
    grid-template-columns: 1fr;
    grid-gap: 16px;
  }
}
