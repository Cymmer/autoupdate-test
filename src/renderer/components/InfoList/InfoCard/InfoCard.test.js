import React from 'react';
import { shallow } from 'enzyme';
import InfoCard from './index';

describe('RENDERING - InfoList > InfoCard', () => {
  it('should render with default props', () => {
    const component = shallow(
      <InfoCard info="Dummy Info" title="Dummy Title" />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render all the syntaxes if there is/are', () => {
    const component = shallow(
      <InfoCard
        info="Dummy Info"
        syntax={['syntax1', 'syntax2']}
        title="Dummy Title"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the actual sample code if there is', () => {
    const component = shallow(
      <InfoCard
        code="Dummy Source Code"
        info="Dummy Info"
        title="Dummy Title"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the output of the sample code if there is a sample code', () => {
    // test first by passing an `output` without a `code`
    let component = shallow(
      <InfoCard info="Dummy Info" output="Dummy Output" title="Dummy Title" />
    );
    expect(component).toMatchSnapshot();

    // then test again by actually passing a `code`
    component = shallow(
      <InfoCard
        code="Dummy Source Code"
        info="Dummy Info"
        output="Dummy Output"
        title="Dummy Title"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the additional information if there is an additional information', () => {
    const component = shallow(
      <InfoCard
        additionalInfo="Dummy Additional Info"
        info="Dummy Info"
        title="Dummy Title"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
