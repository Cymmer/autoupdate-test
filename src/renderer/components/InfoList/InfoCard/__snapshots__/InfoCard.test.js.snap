// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - InfoList > InfoCard should render all the syntaxes if there is/are 1`] = `
<Fragment>
  <Card
    className="InfoCard"
    id={null}
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
  >
    <div
      className="InfoCard_accent"
    />
    <div
      className="InfoCard_content"
    >
      <Text
        className="InfoCard_subject"
        colorClass="n___700"
        element={null}
        id={null}
        type="heading___xs"
      >
        Dummy Title
      </Text>
      <Text
        className="InfoCard_description"
        colorClass="n___600"
        element={null}
        id={null}
        type="body___md"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dummy Info",
            }
          }
        />
      </Text>
      <div
        className="InfoCard_syntax"
      >
        <Text
          className={null}
          colorClass="n___500"
          element={null}
          id={null}
          type="body___md"
        >
          Syntax:
        </Text>
        <Connect(Code)
          className="InfoCard_syntax_code"
          key="0"
          language="text"
        >
          syntax1
        </Connect(Code)>
        <Connect(Code)
          className="InfoCard_syntax_code"
          key="1"
          language="text"
        >
          syntax2
        </Connect(Code)>
      </div>
    </div>
  </Card>
</Fragment>
`;

exports[`RENDERING - InfoList > InfoCard should render the actual sample code if there is 1`] = `
<Fragment>
  <Card
    className="InfoCard"
    id={null}
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
  >
    <div
      className="InfoCard_accent"
    />
    <div
      className="InfoCard_content"
    >
      <Text
        className="InfoCard_subject"
        colorClass="n___700"
        element={null}
        id={null}
        type="heading___xs"
      >
        Dummy Title
      </Text>
      <Text
        className="InfoCard_description"
        colorClass="n___600"
        element={null}
        id={null}
        type="body___md"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dummy Info",
            }
          }
        />
      </Text>
      <div
        className="InfoCard_code"
      >
        <Connect(Code)
          language="text"
        >
          Dummy Source Code
        </Connect(Code)>
      </div>
    </div>
  </Card>
</Fragment>
`;

exports[`RENDERING - InfoList > InfoCard should render the additional information if there is an additional information 1`] = `
<Fragment>
  <Card
    className="InfoCard"
    id={null}
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
  >
    <div
      className="InfoCard_accent"
    />
    <div
      className="InfoCard_content"
    >
      <Text
        className="InfoCard_subject"
        colorClass="n___700"
        element={null}
        id={null}
        type="heading___xs"
      >
        Dummy Title
      </Text>
      <Text
        className="InfoCard_description"
        colorClass="n___600"
        element={null}
        id={null}
        type="body___md"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dummy Info",
            }
          }
        />
      </Text>
    </div>
  </Card>
  <Card
    className="InfoCard_additional"
    id={null}
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
  >
    <Icon
      className="InfoCard_additional_icon"
      icon="info"
      id={null}
      isPulsating={false}
      style={null}
    />
    <Text
      className="InfoCard_additional_text"
      colorClass="n___700"
      element={null}
      id={null}
      type="body___md"
    >
      Dummy Additional Info
    </Text>
  </Card>
</Fragment>
`;

exports[`RENDERING - InfoList > InfoCard should render the output of the sample code if there is a sample code 1`] = `
<Fragment>
  <Card
    className="InfoCard"
    id={null}
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
  >
    <div
      className="InfoCard_accent"
    />
    <div
      className="InfoCard_content"
    >
      <Text
        className="InfoCard_subject"
        colorClass="n___700"
        element={null}
        id={null}
        type="heading___xs"
      >
        Dummy Title
      </Text>
      <Text
        className="InfoCard_description"
        colorClass="n___600"
        element={null}
        id={null}
        type="body___md"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dummy Info",
            }
          }
        />
      </Text>
    </div>
  </Card>
</Fragment>
`;

exports[`RENDERING - InfoList > InfoCard should render the output of the sample code if there is a sample code 2`] = `
<Fragment>
  <Card
    className="InfoCard"
    id={null}
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
  >
    <div
      className="InfoCard_accent"
    />
    <div
      className="InfoCard_content"
    >
      <Text
        className="InfoCard_subject"
        colorClass="n___700"
        element={null}
        id={null}
        type="heading___xs"
      >
        Dummy Title
      </Text>
      <Text
        className="InfoCard_description"
        colorClass="n___600"
        element={null}
        id={null}
        type="body___md"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dummy Info",
            }
          }
        />
      </Text>
      <div
        className="InfoCard_code"
      >
        <Connect(Code)
          language="text"
        >
          Dummy Source Code
        </Connect(Code)>
        <Text
          className={null}
          colorClass="n___700"
          element={null}
          id={null}
          type="heading___xxxs"
        >
          Output:
        </Text>
        <Connect(Code)>
          Dummy Output
        </Connect(Code)>
      </div>
    </div>
  </Card>
</Fragment>
`;

exports[`RENDERING - InfoList > InfoCard should render with default props 1`] = `
<Fragment>
  <Card
    className="InfoCard"
    id={null}
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
  >
    <div
      className="InfoCard_accent"
    />
    <div
      className="InfoCard_content"
    >
      <Text
        className="InfoCard_subject"
        colorClass="n___700"
        element={null}
        id={null}
        type="heading___xs"
      >
        Dummy Title
      </Text>
      <Text
        className="InfoCard_description"
        colorClass="n___600"
        element={null}
        id={null}
        type="body___md"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dummy Info",
            }
          }
        />
      </Text>
    </div>
  </Card>
</Fragment>
`;
